lib/rubocop/cop/style/hash_each_methods.rb

```diff
@@ -38,28 +38,55 @@ class HashEachMethods < Base
           ({block numblock} $(send (send _ ${:keys :values}) :each) ...)
         PATTERN
 
+        # @!method block_pass_each(node)
+        def_node_matcher :block_pass_each, <<~PATTERN
+          (send $(send _ ${:keys :values}) :each (block_pass (sym _)))
+        PATTERN
+
         def on_block(node)
-          register_kv_offense(node)
+          kv_each(node) do |target, method|
+            register_kv_offense(target, method)
+          end
         end
 
         alias on_numblock on_block
 
+        def on_block_pass(node)
+          block_pass_each(node.parent) do |target, method|
+            register_block_pass_offense(node, target, method)
+          end
+        end
+
         private
 
-        def register_kv_offense(node)
-          kv_each(node) do |target, method|
-            parent_receiver = target.receiver.receiver
-            return unless parent_receiver
-            return if allowed_receiver?(parent_receiver)
+        def register_kv_offense(target, method)
+          return if ignored_receiver?(target)
 
-            msg = format(message, prefer: "each_#{method[0..-2]}", current: "#{method}.each")
+          add_offense(kv_range(target), message: format_message(method)) do |corrector|
+            correct_key_value_each(target, corrector)
+          end
+        end
 
-            add_offense(kv_range(target), message: msg) do |corrector|
-              correct_key_value_each(target, corrector)
-            end
+        def register_block_pass_offense(node, target, method)
+          return if ignored_receiver?(node.parent)
+
+          range = target.loc.selector.with(end_pos: node.parent.loc.selector.end_pos)
+          add_offense(range, message: format_message(method)) do |corrector|
+            corrector.replace(range, "each_#{method[0..-2]}")
           end
         end
 
+        def ignored_receiver?(node)
```

Comment:
Can you inline this method into `allowed_receiver?(receiver)? Because the terminologies and roles are mixed.

Reply:
Indeed it is. I inlined it.